prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,6 +208,87 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Get a sortable list of facilities for multiple deliveries
+     *
+     * @param request HTML request body.
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getDeliveryFacilitiesBulk")
+    fun getDeliveryFacilitiesBulk(
+        @HttpTrigger(
+            name = "getDeliveryFacilitiesBulk",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{receiver}/facilities"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("receiver") receiver: String
+    ): HttpResponseMessage {
+        // Do authentication
+        val claims = AuthenticatedClaims.authenticate(request)
+            ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+
+        val userOrgName = this.validateOrgSvcName(receiver)
+            ?: return HttpUtilities.notFoundResponse(
+                request,
+                "$receiver: invalid organization or service identifier"
+            )
+
+        // Authorize based on: org name in the path == org name in claim.  Or be a prime admin.
+        if (!deliveryFacade.checkAccessAuthorizationForOrg(claims, userOrgName, null, request)) {
```

Comment:
Please resolve this or otherwise comment why it was not resolved.

Reply:
We can't use that method because of the scope from the ReportFileFacade. We'd have to rework some of the inherited code and that would be a bit out of scope.

Reply:
```suggestion
        if (!claims.authorizedForSendOrReceive(userOrgName, null, request )) {
```

I think this should work